File: HW3/task3.cpp
#include <iostream>
#include <math.h>

int NUM_ITERATION = 999;

long double root_and_square(long double value) {
  long double eps, left, right, mid;

  eps = 0.0000001;
  left = 1;
  right = pow(value,0.5);
  for (int i = 0; i < NUM_ITERATION; i++) {
    mid = left + (right - left) / 2;
    if (((pow(mid,2) + pow(mid,0.5)) < value + eps) &&
		((pow(mid,2) + pow(mid, 0.5)) > value - eps)) {
	  return right;
    }
    else if ((pow(mid,2) + pow(mid, 0.5)) > value - eps) {
      right = mid;
    }
    else {
      left = mid;
    }
  }
  return right;
}


int main() {
  long double c;

  std::cin >> c;
  std::cout << root_and_square(c);
  return (0);
}

File: HW3/task3.py
def root_and_square(value):
    eps = 0.0000001
    left = 1
    right = value**0.5
    for i in range(999):
        mid = left + (right - left) / 2
        if ((mid**2 + mid**0.5) < value + eps) and ((mid**2 + mid**0.5) > value - eps):

            return right
        elif mid**2 + mid**0.5 > value + eps:
            right = mid
        else:
            left = mid
    return right

if __name__ == "__main__":
    c = float(input())
    print(root_and_square(c))